tests/exceptions/exceptions_constructors.cpp

```diff
@@ -27,7 +27,7 @@ inline void check_exception(sycl::exception e, const std::error_code errcode,
   CHECK(e.category() == errcat);
   CHECK(e.what() != nullptr);
   CHECK(e.has_context() == false);
-  CHECK_THROWS_MATCHES([&e] { e.get_context(); }, sycl::exception,
+  CHECK_THROWS_MATCHES([&e] { e.get_context(); }(), sycl::exception,
```

Comment:
But if we are using macros, why using lambdas?
```suggestion
  CHECK_THROWS_MATCHES(e.get_context(), sycl::exception,
```
?

Reply:
Thanks, I haven't thought about it. I've applied the suggestion in 3106ff5980d406e1dd424a01db4d93b6a38f7754